### PR TITLE
[Teams] Fix Teams menu rendering even when there are backend errors

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

In https://github.com/gitpod-io/gitpod/issues/6035 and https://github.com/gitpod-io/gitpod/issues/6037 we noticed that:

- [x] When a user account gets deleted, their team memberships remain active, causing errors for all other team members (now fixed in this PR ✅)

- [x] When getting the members of one of your teams fail, the Team menu rendering is broken (all members count stay as `...`, and the tabs menu is not the Team menu but the User menu) (also fixed in this PR ✅)

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/6035
Fixes https://github.com/gitpod-io/gitpod/issues/6037

## How to test

1. Create two users, and add them to a Team
2. Delete one of the user accounts (without manually removing them from the Team)
3. The other user should still have a functional Teams dropdown menu, showing the Team with only 1 member (themselves)

## Release Notes

```release-note
[Teams] Fix Teams menu rendering even when there are backend errors
```

/uncc

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankeromnes** after the PR has been reviewed.

Associated issue: *#6035*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/user/user-deletion-service.ts
Comment (AlexTugarev) [inline components/server/src/user/user-deletion-service.ts:140]: 👍🏻

Comment (jankeromnes) [reply components/server/src/user/user-deletion-service.ts:140]: @AlexTugarev Did you mean to `/lgtm` this Pull Request? 😇

**Comment by roboquat** [comment]
@JanKoehnlein: Cannot schedule issue - issue does not belong to a team. Use /team to specify one.

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/6041#issuecomment-934337200):

>/schedule
>


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by jankeromnes** [comment]
@JanKoehnlein shouldn't we add only issues to groundwork, not PRs?

**Review by gtsiolis — APPROVED** [review]
UX works like a charm! 🔮 

Thanks @jankeromnes for fixing this and always adding steps for how to test changes in the PR description. Makes everything so much better, reduces cognitive load on the reviewer side, and allows reviewers not familiar with a product feature to easily dive into a review with more confidence. 🍊 

Approving to unblock merging but holding in case @AlexTugarev wants to take another look (see https://github.com/gitpod-io/gitpod/pull/6041#pullrequestreview-771218756) or @JanKoehnlein wants to make any more changes for testing groundwork automation (see https://github.com/gitpod-io/gitpod/pull/6041#event-5410203045).

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 0ebb657213e2dc70a421d88d06208fa0b6497853</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6041#pullrequestreview-771388239" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankeromnes** after the PR has been reviewed.

Associated issue: *#6035*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
works as advertised! thanks!

/approve

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6041#issuecomment-934421113" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6041#pullrequestreview-771388239" title="Approved">gtsiolis</a>*

Associated issue: *#6035*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
